pkg/api/apiloader.go

```diff
@@ -41,6 +42,12 @@ func (a *Apiloader) DeserializeContainerService(contents []byte, validate bool)
 // LoadContainerService loads an ACS Cluster API Model, validates it, and returns the unversioned representation
 func (a *Apiloader) LoadContainerService(contents []byte, version string, validate bool) (*ContainerService, error) {
 	switch version {
+	case v20170831.APIVersion:
```

Comment:
At some point it might make sense to move this to it's separate function like: 
`func (a *Apiloader) LoadAgentPoolOnlyContainerService(contents []byte, version string, validate bool) (*ContainerService, error) { 
...
}
`
The reason is that the deserialized types of the input (`contents`) are different and both containerservice and agentpoolonly type could have same API version depending on release schedule.